### ISSUE TITLE
Remember playing queue and last playing position

### ISSUE THREAD

**Issue by ElleshaHackett** [deluan/navidrome#245]
I was thinking about audiobook support. Audiobooks don't necessarily have to be very different from normal albums. But tracks can be fairly long. I think it would be a great feature to remember the playing queue and especially (for audiobooks) the last playing position even after a page refresh or after relogin after a few days.

This could be at least in the web player and possibly this info could be exposed in an API (compatible format with other players if possible) too.

**Comment by deluan** [comment]
Yes, and this will be implemented in the Subsonic API first: [Bookmarks](http://www.subsonic.org/pages/api.jsp#getBookmarks), and [SavePlayQueue](http://www.subsonic.org/pages/api.jsp#savePlayQueue). 

It's already on _my_ queue :)

**Comment by diyfr** [comment]
Can you memorize the volume level (web-ui)? 
@deluan Congratulations again for this project

**Comment by deluan** [comment]
This is now supported through the Subsonic API, so it is ready to use by Subsolic clients: https://github.com/deluan/navidrome/pull/425

Next is implementing it in the Web UI

**Comment by obbardc** [comment]
any plans for finishing the Web UI part?
i'd enjoy this for audiobooks ;-)

**Comment by deluan** [comment]
Don't have a timeline for this, but it is in my todo list for sure

**Comment by DocBrown101** [comment]
Is there anything new yet?

The app Ultrasonic in the PlayStore already supports bookmarks. Unfortunately, this does not work with this server.

**Comment by deluan** [comment]
@DocBrown101 This issue is still open only to track the Web UI part, but it is already implemented in the Subsonic API. 

Re: not working in Ultrasonic, a bug in the JSON format was found this week (#1099) preventing Ultrasonic to work properly. 

The fix is already in place in the development builds, and will be part of next release (0.43.0), at the end of the month

**Comment by zeezhao** [comment]
Hi. This is still a nice-to-have feature i.e. if user or server session was terminated, next time it runs it remembers and starts from last known playing position.

Thanks

**Comment by vurtomatic** [comment]
+1

Also looking for "last play position" feature because I listen to mixes that are typically 2hrs.

But since the API has been implemented, does this mean Subsonic clients can already do this? I'm using [Supersonic](https://github.com/dweymouth/supersonic) and I'm not sure if it does.

**Comment by bornav** [comment]
The "last play position" would be nice, but from my testing, the current player does not fully support setting the play position(at-leas i could not figure it out)

**Comment by quite** [comment]
For me, remembering the queue and the currently playing file across (accidental) logout, re-loging, page reload etc. is the most important. But I see the need for remembering position in currently playing file, not in the least for audiobooks.

**Comment by JaneX8** [comment]
I still really would like to see this feature too. This would instantly make Navidrome suitable for audiobooks.

**Comment by bornav** [comment]
@quite there is a working image for that (up to date with the pr listed above), been using it for ~year, if you want to test it.
BREAKING changes to the database, so don't use it on a db you care about, or make backups
bornav/navidrome-bornav-test-cloud-sync

**Comment by bornav** [comment]
db revert and deploy commands are in code, but am not confident enough to tell you to use it, since did not follow the code changes from the repo since the PR was opened(don't know if any breaking changes were introduced),
you can always make a backup, deploy, and see if broken

**Comment by sprnza** [comment]
> also not fammiliar with any clients that support the queue, so it was tested with the webui, if you use something else this solution is not certain to work

@bornav It looks like [this client](https://f-droid.org/en/packages/com.cappielloantonio.notquitemy.tempo/) supports this feature. Could you please check it against your image?

**Comment by bornav** [comment]
@sprnza Thank you for the suggestion, tested with it and it is ether not working or im using it wrong, will investigate further

have updated the branch with main, so it should be now safe to test with latest databases

**Comment by bornav** [comment]
on further testing it does work, one way when involving the webui, sync from web will be synced to the app but not the other way
further testing required, preferably across other confirmed working clients
